lib/rules/meta-property-ordering.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @fileoverview Enforces the order of meta properties
+ */
+
+'use strict';
+
+const { getKeyName, getRuleInfo } = require('../utils');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Enforces the order of meta properties',
+      category: 'Rules',
+      recommended: false,
+    },
+    type: 'suggestion',
+    fixable: 'code',
+    schema: [{
+      type: 'array',
+      elements: { type: 'string' },
+    }],
+  },
+
+  create (context) {
+    const sourceCode = context.getSourceCode();
+    const info = getRuleInfo(sourceCode.ast);
+
+    const message = 'The meta properties should be placed in a consistent order: [{{order}}].';
+    const order = context.options[0] || ['type', 'docs', 'fixable', 'schema', 'messages'];
+
+    const orderMap = new Map(order.map((name, i) => [name, i]));
+
+    return {
+      Program () {
+        if (
+          !info ||
+          !info.meta ||
+          info.meta.properties.length < 2
+        ) {
+          return;
+        }
+
+        const propsActual = info.meta.properties
+          .filter(prop => orderMap.has(getKeyName(prop)));
+
+        for (let i = 1, j = propsActual.length; i < j; i += 1) {
```

Comment:
nits: can be `i < propsActual.length`?
 is `j` used somewhere?

Reply:
I saw this pattern multiple times in the codebase of `eslint-plugin-react`. I think people write this for a potential performance gain. Since I don't know the truth about this, I will change it as requested.

Reply:
This is outdated after last rewrite.